File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/StringInputFieldView.java
package com.saizad.mvvm.components.form.ui;

import android.content.Context;
import android.util.AttributeSet;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class StringInputFieldView extends InputFieldView<String> {

    public StringInputFieldView(@NotNull Context context) {
        super(context);
    }

    public StringInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public StringInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public String resolveFrom(@NotNull CharSequence charSequence) {
        if(charSequence.toString().isEmpty()){
            return null;
        }
        return charSequence.toString();
    }

    @Override
    public CharSequence resolve(@Nullable String value) {
        return value;
    }

    @Override
    public boolean isSame(String value, String prevValue) {
        return value.equals(prevValue);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/StringInputFieldView.kt
package com.saizad.mvvm.components.form.ui

import android.content.Context
import android.util.AttributeSet

open class StringInputFieldView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : InputFieldView<String>(context, attrs, defStyleAttr) {

    override fun resolveFrom(charSequence: CharSequence): String? {
        return if (charSequence.toString().isEmpty()) {
            null
        } else charSequence.toString()
    }

    override fun resolve(value: String?): CharSequence? {
        return value
    }

    override fun isSame(value: String, prevValue: String): Boolean {
        return value == prevValue
    }
}